webapp/src/ts/modals/edit-user/update-password.component.ts

```diff
@@ -48,13 +53,27 @@ export class UpdatePasswordComponent extends EditUserAbstract implements OnInit
     this.errors = {};
     this.setProcessing();
     if (this.validatePasswordFields()) {
-      const updates = { password: this.editUserModel.password };
+      const password = this.editUserModel.password;
+      const updates = { password };
       const username = this.editUserModel.username;
       this.updateUserService
         .update(username, updates, username, this.editUserModel.currentPassword)
         .then(() => {
-          this.setFinished();
-          this.windowReload();
+          const data = this.getLoginData(username, password);
+          this.userLoginService
+            .login(data)
+            .catch(err => {
+              if (err.status === 302) {
+                this.setFinished();
+                this.close();
+
+                this.modalService
+                  .show(ConfirmPasswordUpdatedComponent)
+                  .catch(() => this.windowReload());
```

Comment:
If you want to trigger the reload, no matter what action the user does: clicking ok or on the top-right x or outside the modal, you need to change this to:
```
this.modalService
  .show(ConfirmPasswordUpdatedComponent)
  .catch(() => {})
  .finally(() => this.windowReload());
```

Reply:
Done, thanks.